flink-python/pyflink/table/table_environment.py

```diff
@@ -1695,12 +1695,14 @@ def _add_jars_to_j_env_config(self, config_key):
         jar_urls = self.get_config().get_configuration().get_string(config_key, None)
         if jar_urls is not None:
             # normalize and remove duplicates
-            jar_urls_set = set([jvm.java.net.URL(url).toString() for url in jar_urls.split(";")])
+            # do not use set, because set does not guarantee order.
+            jar_urls_list = [jvm.java.net.URL(url).toString() for url in jar_urls.split(";")]
```

Comment:
I guess this is not necessary. Why we need to keep the order?

Reply:
> I guess this is not necessary. Why we need to keep the order?

+1, because the empty string is impossible in the middle of set. Therefore it's unnecessary to update this.

Reply:
In normal circumstances, keeping the order of classpaths is not necessary.

When investigating some problem, the random order is disturbing,
we spent many times to find why the classpaths is changed randomly.
and list is efficient enough in this case. So I think keep order 
is better choice. Keep things stable, make things simple.